packages/workflows/scripts/build.mjs

```diff
@@ -0,0 +1,183 @@
+// Get each definition.json file in the directories of the src folder
+// Check if there is a node in the nodes array with the definition_id of "caido/code-js"
+// If there is, get the alias of the node, and check if there is a file in the folder with the same name as the alias
+// If there is, build the file using vitejs build API and copy the build output to dist-workflows/<workflow_name>/<alias>.js
+
+import { build } from "vite";
+import path from "path";
+import fs from "fs";
+import { fileURLToPath } from "url";
+import { builtinModules } from "module";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const rootDir = path.join(__dirname, "..");
+const distDir = path.join(rootDir, "..", "..", "dist-workflows");
+
+/**
+ * Builds a file using vitejs build API
+ * @param {string} entrypoint - The path to the entrypoint file
+ * @param {string} workflowDir - The path to the workflow directory
+ * @returns {string} The path to the dist file
+ */
+const buildFile = async (entrypoint, workflowDir) => {
+    const outDir = path.join(distDir, path.basename(workflowDir));
+
+    await build({
+        build: {
+            lib: {
+                entry: entrypoint,
+                fileName: (_, entryName) => `${entryName}.js`,
+                formats: ["es"],
+            },
+            outDir,
+            rollupOptions: {
+                external: [/caido:.+/, ...builtinModules],
+                output: {
+                    manualChunks: undefined,
+                },
+            },
+        },
+    });
+
+
+    const fileName = path.basename(entrypoint).split(".")[0];
+    return path.join(outDir, `${fileName}.js`);
+}
+
+/**
+ * Read the workflow definition file of a workflow directory
+ * @param {string} workflowDir - The path to the workflow directory
+ * @returns {Object} The workflow definition
+ */
+const readWorkflowDefinition = (workflowDir) => {
+    const definitionPath = path.join(rootDir, "src", workflowDir, "definition.json");
+    try {
+        return JSON.parse(fs.readFileSync(definitionPath, "utf8"));
+    } catch (e) {
+        throw new Error(`[!] Failed to read ${definitionPath}`);
+    }
+}
+
+/**
+ * Read the workflow manifest file of a workflow directory
+ * @param {string} workflowDir - The path to the workflow directory
+ * @returns {Object} The workflow manifest
+ */
+const readWorkflowManifest = (workflowDir) => {
+    const manifestPath = path.join(rootDir, "src", workflowDir, "manifest.json");
+    try {
+        return JSON.parse(fs.readFileSync(manifestPath, "utf8"));
+    } catch (e) {
+        throw new Error(`[!] Failed to read ${manifestPath}`);
+    }
+}
+
+/**
+ * Validates manifest
+ * @param {string} workflowDir - The path to the workflow directory
+ * @param {Object} definition - The workflow definition
+ * @param {Object} manifest - The workflow manifest
+ */
+const validateManifest = (workflowDir, definition, manifest) => {
```

Comment:
I would look into using zod to validate your manifest
https://zod.dev/?id=basic-usage

You basically define the structure of your manifest in a zod schema, then call `schema.parse(manifest)` to validate it